README.md

```diff
@@ -2,13 +2,84 @@ harvester-installer
+   and container images.
+3. Create the `harvester-cluster-repo` container image, which provides
+   a helm repository including the charts from the previous step.
+4. Package everything from the above steps into an ISO image.  The ISO
+   image is built using [Elemental Toolkit](https://github.com/rancher/elemental-toolkit/),
+   and is based on [harvester/os2](https://github.com/harvester/os2),
+   which in turn is based on [SLE Micro](https://www.suse.com/products/micro/).
+
+The built ISO image is written to the `dist/artifacts` directory.
+
+## Harvester Installation Process
+
+Harvester can be installed by either [booting the Harvester ISO](https://docs.harvesterhci.io/v1.2/install/index/),
+or via [PXE Boot](https://docs.harvesterhci.io/v1.2/install/pxe-boot-install).
+When booting via ISO, `harvester-installer` runs interactively on the
+system console to allow you to configure the system.  When booting via
+PXE, you don't get the interactive installer - instead you need to
+provide YAML files specifying the configuration to apply.
+
+In both cases (ISO boot and PXE boot), the `harvester-installer` binary
+still _runs_ in order to provision the system.  This is put in place by
+[system/oem/91_installer.yaml](https://github.com/harvester/harvester-installer/blob/master/package/harvester-os/files/system/oem/91_installer.yaml)
+which in turn calls [setup-installer.sh](https://github.com/harvester/harvester-installer/blob/master/package/harvester-os/files/usr/bin/setup-installer.sh)
+to start the installer on tty1.
+
+When booted via ISO, the installer will prompt for configuration
+information (create a new cluster / join an existing cluster, what
+disks to use, network config, etc.).  When booted via PXE, the kernel
+command line parameter `harvester.install.automatic=true` causes the
+interactive part to be skipped, and config will be retrieved from the
+URL specified by `harvester.install.config_url`.
+
+Either way (ISO or PXE), the installer writes the final config out to
+a temporary file which is passed to [harv-install](https://github.com/harvester/harvester-installer/blob/master/package/harvester-os/files/usr/sbin/harv-install)
+which in turn calls `elemental install` to provision the system.
+The harv-install script also preloads all the container images.
+Finally the system is rebooted.
+
+On the newly installed system, `harvester-installer` remains active
+on the console in order to show the cluster management URL along with
+the current node's hostname and IP address.
+
+## Hacking the Interactive Part of `harvester-installer`
+
+Ordinarily `harvester-installer` needs to be run from a booted ISO
+so it can actually install a system.  But, if you're only working
+on changes to the interactive part of the installer (e.g. adding
+or changing fields, or altering the workflow) and don't actually
+need to perform final installation, the binary can be quickly tested
+using vagrant with vagrant-libvirt.  Here's how:
```

Comment:
just have the idea in my mind today, will it be easier to test the executable in a container?

Reply:
There's two issues I can think of when running it in a container:
1. It lists disks to install on in the second screen. With a VM we can easily give it a 500G disk to show in that list. I'm not sure how to do the same thing in a container (although I assume there's a way).
2. Part way through the installation, after you select a NIC, it actually applies the network configuration to the system, which I don't think is going to work in a container (so you wouldn't be able to proceed past that screen).

Reply:
Thanks for the detailed info. A VM is a better isolate environment here.